src/prepare_pcap.c

```diff
@@ -101,17 +98,77 @@ size_t get_ethertype_offset(int link, const uint8_t* pktdata)
         /* http://www.tcpdump.org/linktypes/LINKTYPE_LINUX_SLL.html */
         /* pkttype[2], arphrd_type[2], lladdrlen[2], lladdr[8], ethertype[2] */
         offset = 14;
+    } else if (link == DLT_IEEE802_11_RADIO) {
+        /* get RadioTap header length
+        */
+        uint16_t radio_hdr_len = 0;
+        /* http://www.radiotap.org */
+        /* rdtap_version[1], pad[1], rdtap_hdr_len[2], rdtap_flds[4] */
+        memcpy(&radio_hdr_len, pktdata + 2, sizeof(radio_hdr_len));
+#ifdef __HPUX
+        /* http://radiotap.org */
+        /* all data fields in the radiotap header are to be specified in little endian byte-order
+        */
+        radio_hdr_len = htons(radio_hdr_len);
+#endif
+        offset += radio_hdr_len;
+        /* 2 bits */
+        uint8_t frame_type = 0;
+        /* 4 bits */
+        uint8_t frame_sub_type = 0;
+        /* Frame Control Field */
+        uint16_t frame_ctl_fld;
+        memcpy(&frame_ctl_fld, pktdata + radio_hdr_len, sizeof(frame_ctl_fld));
+        /* extract frame type and subtype from Frame Control Field
+        */
+        frame_ctl_fld = ntohs(frame_ctl_fld);
+        frame_type = frame_sub_type = frame_ctl_fld & 0x00FF;
+        frame_type <<= 4;
+        frame_type >>= 6;
```

Comment:
It is. I just thought breaking it down this way would make it easier to understand for some.
There is room for improvements and any remarks are welcome as this is a pretty raw implementation.